types/mydecimal_test.go

```diff
@@ -135,30 +136,57 @@ func (s *testMyDecimalSuite) TestFromFloat(c *C) {
 
 func (s *testMyDecimalSuite) TestToFloat(c *C) {
 	tests := []struct {
-		s string
-		f float64
+		in  string
+		out string
+		err error
 	}{
-		{"12345", 12345},
-		{"123.45", 123.45},
-		{"-123.45", -123.45},
-		{"0.00012345000098765", 0.00012345000098765},
-		{"1234500009876.5", 1234500009876.5},
-		{"1e39", 1e39},
-		{"1e-39", 1e-39},
-		{"1e00", 1},
-		{"1e001", 10},
-		{"-9223372036854775807", -9223372036854775807},
-		{"-9223372036854775808", -9223372036854775808},
-		{"18446744073709551615", 18446744073709551615},
-		{"123456789.987654321", 123456789.987654321},
+		{"12345", "12345", nil},
+		{"123.45", "123.45", nil},
+		{"-123.45", "-123.45", nil},
+		{"0.00012345000098765", "0.00012345000098765", nil},
+		{"1234500009876.5", "1234500009876.5", nil},
+		{"1e39", "1e39", nil},
+		{"1e-39", "1e-39", nil},
+		{"1e00", "1", nil},
+		{"1e001", "10", nil},
+		{"-9223372036854775807", "-9223372036854775807", nil},
+		{"-9223372036854775808", "-9223372036854775808", nil},
+		{"18446744073709551615", "18446744073709551615", nil},
+		{"123456789.987654321", "123456789.987654321", nil},
+		// Testcases from https://github.com/golang/go/blob/master/src/strconv/atof_test.go#L24
+		{"1", "1", nil},
+		{"+1", "1", nil},
+		{"1e23", "1e+23", nil},
+		{"1E23", "1e+23", nil},
+		{"100000000000000000000000", "1e+23", nil},
+		{"123456700", "1.234567e+08", nil},
+		{"99999999999999974834176", "9.999999999999997e+22", nil},
+		{"100000000000000000000001", "1.0000000000000001e+23", nil},
+		{"100000000000000008388608", "1.0000000000000001e+23", nil},
+		{"100000000000000016777215", "1.0000000000000001e+23", nil},
+		{"100000000000000016777216", "1.0000000000000003e+23", nil},
+		{"-1", "-1", nil},
+		{"-0.1", "-0.1", nil},
+		{"-0", "-0", nil},
+		{"1e-20", "1e-20", nil},
+		{"625e-3", "0.625", nil},
+		{"0", "0", nil},
+		{"22.222222222222222", "22.22222222222222", nil},
+		{"1.00000000000000011102230246251565404236316680908203125", "1", nil},
+		{"1.00000000000000011102230246251565404236316680908203124", "1", nil},
+		{"1.00000000000000011102230246251565404236316680908203126", "1.0000000000000002", nil},
+		{"1.00000000000000033306690738754696212708950042724609375", "1.0000000000000004", nil},
+		{"1090544144181609348671888949248", "1.0905441441816093e+30", nil},
+		{"1090544144181609348835077142190", "1.0905441441816094e+30", nil},
 	}
 	for _, ca := range tests {
 		var dec MyDecimal
-		err := dec.FromString([]byte(ca.s))
+		err := dec.FromString([]byte(ca.in))
 		c.Assert(err, IsNil)
 		f, err := dec.ToFloat64()
 		c.Check(err, IsNil)
-		c.Check(f, Equals, ca.f)
+		std, _ := strconv.ParseFloat(ca.out, 64)
```

Comment:
nitpick: please check error. It maybe lead lint fail...

Reply:
fixed